aten/src/ATen/test/extension_backend_test.cpp

```diff
@@ -0,0 +1,52 @@
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/ExtensionBackendRegistration.h>
+
+using namespace at;
+
+int test_int;
+
+Tensor empty_override(IntList size, const TensorOptions & options) {
+  test_int = 1;
+  auto tensor_impl = c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(
+      Storage(
+          caffe2::TypeMeta::Make<float>(), 0, at::DataPtr(), nullptr, false),
+      FPGATensorId(),
+      false);
+  return Tensor(std::move(tensor_impl));
+}
+
+Tensor empty_like_override(const Tensor & self, const TensorOptions & options) {
+  test_int = 2;
+  return self;
+}
+
+Tensor add_override(const Tensor & a, const Tensor & b , Scalar c) {
+  test_int = 3;
+  return a;
+}
+
+TEST(BackendExtensionTest, TestRegisterOp) {
+  EXPECT_ANY_THROW(empty({5, 5}, at::kFPGA));
+  register_extension_backend_op(
+    Backend::FPGA,
+    "empty(IntList size, TensorOptions options) -> Tensor", &empty_override);
```

Comment:
what happens if you ask for the dtype?

Reply:
can you add a check that a.dtype (or whatever the equivalent in C++ is) is valid?